React/Modules/RCTStatusBarManager.m

```diff
@@ -72,35 +70,29 @@ - (dispatch_queue_t)methodQueue
   return dispatch_get_main_queue();
 }
 
-- (void)emitEvent:(NSString *)eventName forNotification:(NSNotification *)notification
+static NSDictionary *RCTExportedFrame()
 {
-  CGRect frame = [notification.userInfo[UIApplicationStatusBarFrameUserInfoKey] CGRectValue];
-  NSDictionary *event = @{
-    @"frame": @{
-      @"x": @(frame.origin.x),
-      @"y": @(frame.origin.y),
-      @"width": @(frame.size.width),
-      @"height": @(frame.size.height),
-    },
-  };
-  [self sendEventWithName:eventName body:event];
+  // The In-Call Status Bar pushes the root view down instead of increasing the size of the StatusBar area
+  // that needs to be accounted for in JS. We can fix that by subtracting the y value of the root view
+  // from the StatusBar's height.
+  CGRect rootFrame = RCTPresentedViewController().view.frame;
```

Comment:
We cannot rely on `RCTPresentedViewController` here, it simply will not work.

Reply:
Do you know of a more reliable way to get the y-offset of the app? We can't rely purely on the `statusBarFrame`'s height because iOS reports 40px for the In-Call Status bar, but if navigators use that value then it'll get pushed down 20px more than it should.

Reply:
Oh, I see what you mean, you want to have statusBarFrame to be relative to your RootView position. Right? The problem here is that you cannot assume that:
 * You have only one RootView at any particular the moment;
 * Your RootView is not moving.

You probably should use `<SafeAreaView>` to get same effect.
I would start from simple API where statusBarFrame always return absolute values.